math/vecops/inc/ROOT/RStrongBool.hxx

```diff
@@ -0,0 +1,54 @@
+// Author: Danilo Piparo CERN  06/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RStrongBool
+#define ROOT_RStrongBool
+
+#include <iostream>
+
+namespace ROOT {
+namespace VecOps {
+
+// clang-format off
+///////////////////////////////////////////////////////////////////////////////
+/// \brief Strongly typed boolean type to prevent casts.
+/// The storage of the value is of type int.
+// clang-format on
+class RStrongBool final {
+public:
+  RStrongBool();
+  ~RStrongBool();
+  template<typename T>
+  RStrongBool(T val) = delete;
+  RStrongBool(bool val);
+  // this is inlined to workaround missing symbols errors
+  explicit operator bool() const
+{
+	return bool(fVal);
+}
```

Comment:
why must the cast be explicit? one would reasonably expect `RStrongBool` to be usable in place of a `bool`